server/service/handler_test.go

```diff
@@ -269,117 +259,232 @@ func TestAPIRoutesConflicts(t *testing.T) {
+		require.Equal(t, http.StatusOK, rr.Code)
+	}
+
+	// get the metrics
+	req := httptest.NewRequest("GET", "/metrics", nil)
+	rr := httptest.NewRecorder()
+	router.ServeHTTP(rr, req)
+	require.Equal(t, http.StatusOK, rr.Code)
+
+	rxMetric := regexp.MustCompile(`^([\w_]+)({[^}]+})? (.+)$`)
+	rxHandler := regexp.MustCompile(`handler="([\w_]+)"`)
+
+	// expected metric names and their counts
+	metricCounts := map[string]int{
+		"go_gc_duration_seconds":                     0,
+		"go_gc_duration_seconds_sum":                 0,
+		"go_gc_duration_seconds_count":               0,
+		"go_goroutines":                              0,
+		"go_info":                                    0,
+		"go_memstats_alloc_bytes":                    0,
+		"go_memstats_alloc_bytes_total":              0,
+		"go_memstats_buck_hash_sys_bytes":            0,
+		"go_memstats_frees_total":                    0,
+		"go_memstats_gc_cpu_fraction":                0,
+		"go_memstats_gc_sys_bytes":                   0,
+		"go_memstats_heap_alloc_bytes":               0,
+		"go_memstats_heap_idle_bytes":                0,
+		"go_memstats_heap_inuse_bytes":               0,
+		"go_memstats_heap_objects":                   0,
+		"go_memstats_heap_released_bytes":            0,
+		"go_memstats_heap_sys_bytes":                 0,
+		"go_memstats_last_gc_time_seconds":           0,
+		"go_memstats_lookups_total":                  0,
+		"go_memstats_mallocs_total":                  0,
+		"go_memstats_mcache_inuse_bytes":             0,
+		"go_memstats_mcache_sys_bytes":               0,
+		"go_memstats_mspan_inuse_bytes":              0,
+		"go_memstats_mspan_sys_bytes":                0,
+		"go_memstats_next_gc_bytes":                  0,
+		"go_memstats_other_sys_bytes":                0,
+		"go_memstats_stack_inuse_bytes":              0,
+		"go_memstats_stack_sys_bytes":                0,
+		"go_memstats_sys_bytes":                      0,
+		"go_threads":                                 0,
+		"http_request_duration_seconds_bucket":       0,
+		"http_request_duration_seconds_sum":          0,
+		"http_request_duration_seconds_count":        0,
+		"http_request_size_bytes_bucket":             0,
+		"http_request_size_bytes_sum":                0,
+		"http_request_size_bytes_count":              0,
+		"http_requests_total":                        0,
+		"http_response_size_bytes_bucket":            0,
+		"http_response_size_bytes_sum":               0,
+		"http_response_size_bytes_count":             0,
+		"process_cpu_seconds_total":                  0,
+		"process_max_fds":                            0,
+		"process_open_fds":                           0,
+		"process_resident_memory_bytes":              0,
+		"process_start_time_seconds":                 0,
+		"process_virtual_memory_bytes":               0,
+		"process_virtual_memory_max_bytes":           0,
+		"promhttp_metric_handler_requests_in_flight": 0,
+		"promhttp_metric_handler_requests_total":     0,
+	}
```

Comment:
Wrote this test first and ran with the old output, in order to identify the breaking changes.